src/libfsm/minimise.c

```diff
@@ -265,7 +265,7 @@ build_minimised_mapping(const struct fsm *fsm,
 				if (next == NO_ID) {
 					continue;    /* 0 elements, skip */
 				}
-				should_gather_EC_labels = next & SMALL_EC_FLAG;
+				should_gather_EC_labels = (next & SMALL_EC_FLAG) != 0;
```

Comment:
I hate when compilers warn about this! but okay.

Reply:
In this specific case it makes sense, because `SMALL_EC_FLAG` is the MSB, `should_gather_EC_labels` is a (signed) `int` being reluctantly used in place of a `bool` due to C89, whereas `next` is an `fsm_state_t`, an unsigned int, so it's implicit signed to unsigned conversion. I suppose we could have also used `!!`.